pages/docs/observability/sdk/typescript/setup.mdx

```diff
@@ -0,0 +1,139 @@
+---
+title: TypeScript SDK - Setup
+description: Tracing setup for the TypeScript SDK.
+label: "Version: JS SDK v4"
+---
+
+# TypeScript SDK - Setup
+
+import EnvJS from "@/components-mdx/env-js.mdx";
+
+The Langfuse TypeScript SDK offers two setup approaches:
+
+1. [**Tracing**](#tracing-setup) for [Langfuse Observability](/docs/observability/overview) using OpenTelemetry
+2. [**Client**](#client-setup) for other Langfuse features like prompt management, evaluation, or accessing the Langfuse API
+
+## Tracing Setup [#tracing-setup]
+
+<Steps>
+
+### Installation
+
+Install the relevant packages for a full tracing setup:
+
+```bash
+npm install @langfuse/tracing @langfuse/otel @opentelemetry/sdk-node
+```
+
+- **`@langfuse/tracing`**: Core tracing functions (`startObservation`, `startActiveObservation`, etc.)
+- **`@langfuse/otel`**: The `LangfuseSpanProcessor` to export traces to Langfuse.
+- **`@opentelemetry/sdk-node`**: The OpenTelemetry SDK for Node.js.
+
+Learn more about the packages [here](/docs/observability/sdk/typescript/overview#packages).
+
+### Register your credentials
+
+Add your Langfuse credentials to your environment variables. Make sure that you have a `.env` file in your project root and a package like `dotenv` to load the variables.
+
+<EnvJS />
+
+### Initialize OpenTelemetry
+
+The Langfuse TypeScript SDK's tracing is built on top of OpenTelemetry, so you need to set up the OpenTelemetry SDK. The `LangfuseSpanProcessor` is the key component that sends traces to Langfuse.
+
+```ts filename="instrumentation.ts" /LangfuseSpanProcessor/
+import { NodeSDK } from "@opentelemetry/sdk-node";
+import { LangfuseSpanProcessor } from "@langfuse/otel";
+
+const sdk = new NodeSDK({
+  spanProcessors: [new LangfuseSpanProcessor()],
+});
+
+sdk.start();
+```
+
+The `LangfuseSpanProcessor` is the key component that sends traces to Langfuse.
+
+For more options to configure the LangfuseSpanProcessor such as masking, filtering, and more, see [the advanced usage](/docs/observability/sdk/typescript/advanced-usage).
+
+You can learn more about setting up OpenTelemetry in your JS environment [here](https://opentelemetry.io/docs/languages/js/getting-started/nodejs/).
+
+<Callout type="info">
+**Next.js users:**
+
+If you are using Next.js, please use the OpenTelemetry setup via the `NodeSDK` described above rather than via `registerOTel` from `@vercel/otel`. This is because [the `@vercel/otel` package does not yet support the OpenTelemetry JS SDK v2](https://github.com/vercel/otel/issues/154) on which the `@langfuse/tracing` and `@langfuse/otel` packages are based.
```

Comment:
we need to be more explicit here with an example, many teams will use this and this is too vague. I'd recommend to publish an faq page and link to it from here and overview and general quickstart "Using NextJS on Vercel? Check out this guide"